src/loaders/common.c

```diff
@@ -258,6 +258,51 @@ int libxmp_test_name(uint8 *s, int n)
 	return 0;
 }
 
+int libxmp_copy_name_for_fopen(char *dest, const char *name, int n)
+{
+	int converted_colon = 0;
+	int i;
+
+	/* libxmp_copy_adjust, but make sure the filename won't do anything
+	 * malicious if it's given to fopen. This should only be used on song files.
+	 */
+	if (!strcmp(name, ".") || strstr(name, "..") || name[0] == '\\' || name[0] == '/')
+		return -1;
+
+
+	for (i = 0; i < n - 1; i++) {
+		uint8 t = name[i];
+		if (!t)
+			break;
+
+		/* Reject non-ASCII symbols as they have poorly defined behavior.
+		 */
+		if (t < 32 || t >= 0x7f)
+			return -1;
+
+		/* Reject anything resembling a Windows-style root path. Allow
+		 * converting a single : to / so things like ST-01:samplename work.
+		 */
+		if (t == ':' && !converted_colon) {
+			if (name[i + 1] == '/' || name[i + 1] == '\\')
+				return -1;
+
+			converted_colon = 1;
+			dest[i] = '/';
```

Comment:
Do we want to do this if we're running on amiga?

Reply:
You're right, on Amiga this should probably be left as-is.

Reply:
Just to be sure, what exactly is the correct way to check for Amiga here?

Reply:
> Just to be sure, what exactly is the correct way to check for Amiga here?

That wolud be a pain.. Some well-known compiler macros are `__amigaos4__`,
`__MORPHOS__`, `__amiga__`, `__AMIGA__`, `__AROS__`

In uhexen2, I have the following in arch_def.h (not all of those checks
may be needed or relevant...) :
```
#elif defined(__MORPHOS__) || defined(__AROS__) || defined(AMIGAOS)	|| \
      defined(__amigaos__) || defined(__amigaos4__) ||defined(__amigados__) || \
      defined(AMIGA) || defined(_AMIGA) || defined(__AMIGA__)

#   if !defined(PLATFORM_AMIGA)
#	define	PLATFORM_AMIGA		1
#   endif
```

Reply:
OK, I added these checks pretty much verbatim to src/common.h (since presumably this macro might be useful for other things as well). On Amiga this should just copy the `:` as usual now (I don't have an Amiga toolchain to test with though right now).

Reply:
I will make the AMIGA checks into a new PR now (out of scope of this one.)
I will be using LIBXMP_AMIGA as the macro.

Reply:
See #203

Reply:
I merged #203 so you can rebase and move forward with this

Reply:
OK, I rebased this branch and changed it to use `LIBXMP_AMIGA`.